From 45921f36a7587642e1597ff7767269178a86350e Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 05 Jul 2022 09:10:24 +0300
Subject: [PATCH] [SW-0] rom: update pcie id in recovery

A dedicated pcie dev id exists for single die topology devices in
gaudi3. This patch updates the same in rom recovery flow during
pcie linkup in rom recovery.

Change-Id: I54b0c357068e381a613d4067df8e597f22615d8a
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 0a51b3d..effbaa7 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -315,13 +315,21 @@
 void pcie_set_pcie_device_vendor_id(void)
 {
 	uint32_t id;
+	enum hl_pci_ids pcie_dev_id;
 
-	if (!arcboot_rom_global.die_id)
-		return;
+	if (!arcboot_rom_global.die_id) {
+		if (!arcboot_rom_global.sku)
+			/* Dual die mode. Default pcie dev id can be used */
+			return;
+
+		pcie_dev_id = PCI_IDS_GAUDI3_SINGLE_DIE;
+	} else {
+		pcie_dev_id = PCI_IDS_GAUDI3_DIE1;
+	}
 
 	id = ((PCI_VENDOR_ID_HABANALABS <<
 		D1_PCIE_DBI_SIG_DEVICE_ID_VENDOR_ID_REG_PCI_TYPE0_VENDOR_ID_S) |
-		(PCI_IDS_GAUDI3_DIE1 <<
+		(pcie_dev_id <<
 		D1_PCIE_DBI_SIG_DEVICE_ID_VENDOR_ID_REG_PCI_TYPE0_DEVICE_ID_S));
 
 	regmap_writel(id, mmD0_PCIE_DBI_SIG_DEVICE_ID_VENDOR_ID_REG);
